### DISCUSSION TITLE
2D overlays

### DISCUSSION THREAD

**Discussion by dreamsinmotion** [splinetool/react-spline#36] [Ideas]
Hey, it would be nice to be able to create 2D overlays using objects or null objects as the base reference relative to the camera perspective. For example button components, text, ...

**Comment by marcofugaro** [comment]
Do you maybe have an example of what you wish to accomplish?

**Comment by dreamsinmotion** [comment]
Something like this ThreeJs example, where you can get the coordinates of objects with a projection of the current camera:

https://r105.threejsfundamentals.org/threejs/lessons/threejs-align-html-elements-to-3d.html
https://jsfiddle.net/oy7hqbpf/

**Comment by superguigui** [comment]
@dreamsinmotion in theory you can already access objects in camera and achieve this. But an easier way would probably be to use the new three.js or react-three-fiber exports that allows more control over the scenegraph (but also shift more responsabilities your way).
TLDR: react-spline is a simple player that will display your spline scene as it was in Spline, the r3f and three.js exports are only loading geometries and materials in three.js environments and letting you do what you want with it.

Moving this into discussions :)
